core/loadBalancer/src/whisk/core/loadBalancer/InvokerHealth.scala

```diff
@@ -139,20 +142,32 @@ class InvokerHealth(
             val flattened = ConsulClient.dropKeyLevel(invokerInfo)
             val nested = ConsulClient.toNestedMap(flattened)
 
-            val status = nested map {
+            // Get the new status (some entries may correspond to new instances)
+            val statusMap = nested map {
                 case (key, inner) =>
                     val index = InvokerKeys.extractInvokerIndex(key)
                     val JsString(startDate) = inner(InvokerKeys.startKey).parseJson
                     val JsString(lastDate) = inner(InvokerKeys.statusKey).parseJson
-                    Status(index, startDate, lastDate, isFresh(lastDate), inner(InvokerKeys.activationCountKey).toInt)
+                    (index, Status(index, startDate, lastDate, isFresh(lastDate), inner(InvokerKeys.activationCountKey).toInt))
             }
+            val newStatus = statusMap.toArray.map {_._2} .sortBy(_.index)
```

Comment:
Can be written as `statusMap.values.toArray.sortBy(_.index)` to make the intent of getting the `values` of the map clearer.